tools/src/wyvern/target/corewyvernIL/decl/EffectDeclaration.java

```diff
@@ -0,0 +1,55 @@
+package wyvern.target.corewyvernIL.decl;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+import wyvern.target.corewyvernIL.astvisitor.ASTVisitor;
+import wyvern.target.corewyvernIL.decltype.ConcreteTypeMember;
+import wyvern.target.corewyvernIL.decltype.DeclType;
+import wyvern.target.corewyvernIL.decltype.EffectDeclType;
+import wyvern.target.corewyvernIL.expression.Effect;
+import wyvern.target.corewyvernIL.expression.IExpr;
+import wyvern.target.corewyvernIL.support.TypeContext;
+import wyvern.target.corewyvernIL.type.Type;
+import wyvern.target.corewyvernIL.type.ValueType;
+import wyvern.tools.errors.ErrorMessage;
+import wyvern.tools.errors.FileLocation;
+import wyvern.tools.errors.ToolError;
+
+public class EffectDeclaration extends NamedDeclaration {
+	Effect e;
```

Comment:
make "private"

Also, should this be a Set<Effect> rather than a single Effect?  It should be the same as EffectDeclType in this respect